ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala

```diff
@@ -0,0 +1,92 @@
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package ai.h2o.sparkling.ml.features
+
+import ai.h2o.automl.targetencoding._
+import ai.h2o.sparkling.ml.models.H2OTargetEncoderModel
+import org.apache.spark.h2o.{Frame, H2OContext}
+import org.apache.spark.ml.Estimator
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.util.{DefaultParamsReadable, DefaultParamsWritable, Identifiable}
+import org.apache.spark.sql.{Dataset, SparkSession}
+
+class H2OTargetEncoder(override val uid: String)
+  extends Estimator[H2OTargetEncoderModel]
+  with H2OTargetEncoderBase
+  with DefaultParamsWritable {
+
+  def this() = this(Identifiable.randomUID("H2OTargetEncoder"))
+
+  override def fit(dataset: Dataset[_]): H2OTargetEncoderModel = {
+    val h2oContext = H2OContext.getOrCreate(SparkSession.builder().getOrCreate())
+    val input = h2oContext.asH2OFrame(dataset.toDF())
+    convertRelevantColumnsToCategorical(input)
+    val targetEncoderModel = trainTargetEncodingModel(input)
+    val model = new H2OTargetEncoderModel(uid, targetEncoderModel).setParent(this)
+    copyValues(model)
+  }
+
+  private def trainTargetEncodingModel(trainingFrame: Frame) = try {
+    val targetEncoderParameters = new TargetEncoderModel.TargetEncoderParameters()
+    targetEncoderParameters._withBlending = getBlendedAvgEnabled()
+    targetEncoderParameters._blendingParams = new BlendingParams(getBlendedAvgInflectionPoint(), getBlendedAvgSmoothing())
+    targetEncoderParameters._response_column = getLabelCol()
+    targetEncoderParameters._teFoldColumnName = getFoldCol()
+    targetEncoderParameters._columnNamesToEncode = getInputCols()
+    targetEncoderParameters.setTrain(trainingFrame._key)
+
+    val builder = new TargetEncoderBuilder(targetEncoderParameters)
+    builder.trainModel().get() // Calling get() to wait until the model training is finished.
+    builder.getTargetEncoderModel()
+  } catch {
+    case e: IllegalStateException if e.getMessage.contains("We do not support multi-class target case") =>
+      throw new RuntimeException("The label column can not contain more than two unique values.")
+  }
+
+  override def copy(extra: ParamMap): H2OTargetEncoder = defaultCopy(extra)
+
+
+  //
+  // Parameter Setters
+  //
+  def setFoldCol(value: String): this.type = set(foldCol, value)
+
+  def setLabelCol(value: String): this.type = set(labelCol, value)
+
+  def setInputCols(values: Array[String]): this.type = set(inputCols, values)
+
+  def setHoldoutStrategy(value: H2OTargetEncoderHoldoutStrategy): this.type = set(holdoutStrategy, value)
```

Comment:
Should we use String value here as we discussed right from beginning? See the PR here https://github.com/h2oai/sparkling-water/pull/1359

I in favor of the the string variant